### PR TITLE
Fix ruff discovery for paths containing symbols that can't be represented by the local encoding

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

Fixes https://github.com/astral-sh/ruff-vscode/issues/583

See inline comment for an explanation

## Test Plan

* Created a local directory named `löカひ漢`
* Added a venv
* installed ruff

Before fix: The python script crashed
After: It finds the right ruff binary

### CODE REVIEW

File: bundled/tool/find_ruff_binary_path.py [hunk 1]
```diff
@@ -32,4 +39,4 @@ def find_ruff_binary_path() -> Optional[Path]:
 if __name__ == "__main__":
     ruff_binary_path = find_ruff_binary_path()
     if ruff_binary_path:
-        print(os.fsdecode(str(ruff_binary_path)), flush=True)
+        print(os.fsdecode(ruff_binary_path), flush=True)
```
Comment (MichaReiser) [inline bundled/tool/find_ruff_binary_path.py:42]: I think the call to `str` before calling `os.fsdecode` is just wrong? But how would I know, I don't know nothing of Python

Comment (dhruvmanila) [reply bundled/tool/find_ruff_binary_path.py:44]: Yeah, I think that's correct. `print(<pathlib.Path>)` would use the `pathlib.Path.__str__` method, so the `str` call here is redundant.

**Review by dhruvmanila — APPROVED** [review]
Thanks!

File: bundled/tool/find_ruff_binary_path.py [hunk 2]
```diff
@@ -30,6 +30,13 @@ def find_ruff_binary_path() -> Optional[Path]:
 
 
 if __name__ == "__main__":
+    # Python defaults to the system's local encoding for stdout on Windows.
+    # source: https://docs.python.org/3/library/sys.html#sys.stdout
+    #
+    # But not all paths are representable by the local encoding.
+    # The node process calling this script defaults to UTF8, so let's do the same here.
+    sys.stdout.reconfigure(encoding="utf-8")  # type: ignore We never reconfigure stdout, thus it is guaranteed to not be Any
```
Comment (dhruvmanila) [inline bundled/tool/find_ruff_binary_path.py:38]:
```suggestion
    sys.stdout.reconfigure(encoding="utf-8")  # type: ignore [union-attr] # We never reconfigure stdout, thus it is guaranteed to not be Any
```

**Pull request merged**